cylc/flow/task_state.py

```diff
@@ -154,6 +155,77 @@
 }
 
 
+class RunMode:
+    """The possible run modes of a task/workflow."""
+
+    LIVE = 'live'
+    """Task will run normally."""
+
+    SIMULATION = 'simulation'
+    """Task will run in simulation mode."""
+
+    DUMMY = 'dummy'
+    """Task will run in dummy mode."""
+
+    SKIP = 'skip'
+    """Task will run in skip mode."""
+
+    WORKFLOW = 'workflow'
+    """Default to workflow run mode"""
+
+    MODES = {LIVE, SIMULATION, DUMMY, SKIP, WORKFLOW}
+
+    WORKFLOW_MODES = sorted(MODES - {WORKFLOW})
+    """Workflow mode not sensible mode for workflow.
+
+    n.b. converted to a list to ensure ordering doesn't change in
+    CLI
+    """
+
+    LIVELY_MODES = {LIVE, DUMMY}
+    """Modes which need to have real jobs submitted."""
+
+    GHOSTLY_MODES = {SKIP, SIMULATION}
+    """Modes which completely ignore the standard submission path."""
```

Comment:
Maybe "real" and "jobless" would be clearer?

Reply:
"Jobless" is good, I'm not sure "real" is so good: I think "jobly" might work, or given context `JOB_MODES` and `JOBLESS_MODES`?

Reply:
Also good, because it looks like David S added ghost with an entirely different meaning to the data store manager about 9 months back.

Reply:
`JOBLESS_MODES` and `JOB_MODES` (or even `REAL_JOB_MODES`) look good to me